Doesn't compile for ESP32 Dev Board

In file included from d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/BasicLinearAlgebra.h:10,
                 from C:\Users\cflem\AppData\Local\Temp\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\HowToUse\HowToUse.ino:1:
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:136:43: error: expected unqualified-id before 'const'
  136 |     HorizontalConcat<LeftType, RightType>(const LeftType &l, const RightType &r) : left(l), right(r) {}
      |                                           ^~~~~
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:136:43: error: expected ')' before 'const'
  136 |     HorizontalConcat<LeftType, RightType>(const LeftType &l, const RightType &r) : left(l), right(r) {}
      |                                          ~^~~~~
      |                                           )
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:151:41: error: expected unqualified-id before 'const'
  151 |     VerticalConcat<TopType, BottomType>(const TopType &t, const BottomType &b) : top(t), bottom(b) {}
      |                                         ^~~~~
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:151:41: error: expected ')' before 'const'
  151 |     VerticalConcat<TopType, BottomType>(const TopType &t, const BottomType &b) : top(t), bottom(b) {}
      |                                        ~^~~~~
      |                                         )
In file included from d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/BasicLinearAlgebra.h:181:
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/impl/BasicLinearAlgebra.h: In instantiation of 'BLA::HorizontalConcat<DerivedType, OperandType> BLA::operator||(const MatrixBase<DerivedType, Rows, DerivedType::Cols, DType>&, const MatrixBase<OperandType, Rows, OperandType::Cols, DType>&) [with DerivedType = Matrix<3, 3>; OperandType = Matrix<3, 3>; int Rows = 3; DType = float]':
C:\Users\cflem\AppData\Local\Temp\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\HowToUse\HowToUse.ino:70:39:   required from here
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/impl/BasicLinearAlgebra.h:208:12: error: no matching function for call to 'BLA::HorizontalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >::HorizontalConcat(const BLA::Matrix<3, 3>&, const BLA::Matrix<3, 3>&)'
  208 |     return HorizontalConcat<DerivedType, OperandType>(static_cast<const DerivedType &>(left),
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  209 |                                                       static_cast<const OperandType &>(right));
      |                                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:130:8: note: candidate: 'constexpr BLA::HorizontalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >::HorizontalConcat(const BLA::HorizontalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >&)'
  130 | struct HorizontalConcat : public MatrixBase<HorizontalConcat<LeftType, RightType>, LeftType::Rows,
      |        ^~~~~~~~~~~~~~~~
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:130:8: note:   candidate expects 1 argument, 2 provided
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:130:8: note: candidate: 'constexpr BLA::HorizontalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >::HorizontalConcat(BLA::HorizontalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >&&)'
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:130:8: note:   candidate expects 1 argument, 2 provided
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/impl/BasicLinearAlgebra.h: In instantiation of 'BLA::VerticalConcat<DerivedType, OperandType> BLA::operator&&(const MatrixBase<DerivedType, DerivedType::Rows, Cols, DType>&, const MatrixBase<OperandType, OperandType::Rows, Cols, DType>&) [with DerivedType = Matrix<3, 3>; OperandType = Matrix<3, 3>; int Cols = 3; DType = float]':
C:\Users\cflem\AppData\Local\Temp\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\HowToUse\HowToUse.ino:73:40:   required from here
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/impl/BasicLinearAlgebra.h:218:12: error: no matching function for call to 'BLA::VerticalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >::VerticalConcat(const BLA::Matrix<3, 3>&, const BLA::Matrix<3, 3>&)'
  218 |     return VerticalConcat<DerivedType, OperandType>(static_cast<const DerivedType &>(top),
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  219 |                                                     static_cast<const OperandType &>(bottom));
      |                                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:145:8: note: candidate: 'constexpr BLA::VerticalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >::VerticalConcat(const BLA::VerticalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >&)'
  145 | struct VerticalConcat : public MatrixBase<VerticalConcat<TopType, BottomType>, TopType::Rows + BottomType::Rows,
      |        ^~~~~~~~~~~~~~
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:145:8: note:   candidate expects 1 argument, 2 provided
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:145:8: note: candidate: 'constexpr BLA::VerticalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >::VerticalConcat(BLA::VerticalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >&&)'
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:145:8: note:   candidate expects 1 argument, 2 provided

Tom,

Here's the complete output of the Arduino 2.3.2 IDE with verbose compiler messages:

FQBN: esp32:esp32:esp32
Using board 'esp32' from platform in folder: C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.2
Using core 'esp32' from platform in folder: C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.2

cmd /c if exist "C:\\Users\\cflem\\AppData\\Local\\Temp\\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\\HowToUse\\partitions.csv" COPY /y "C:\\Users\\cflem\\AppData\\Local\\Temp\\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\\HowToUse\\partitions.csv" "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\partitions.csv"
cmd /c if not exist "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\partitions.csv" if exist "C:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\hardware\\esp32\\3.0.2\\variants\\esp32\\partitions.csv" COPY "C:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\hardware\\esp32\\3.0.2\\variants\\esp32\\partitions.csv" "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\partitions.csv"
cmd /c if not exist "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\partitions.csv" COPY "C:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\hardware\\esp32\\3.0.2\\tools\\partitions\\default.csv" "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\partitions.csv"
1 file(s) copied.
cmd /c IF EXIST "C:\\Users\\cflem\\AppData\\Local\\Temp\\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\\HowToUse\\bootloader.bin" ( COPY /y "C:\\Users\\cflem\\AppData\\Local\\Temp\\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\\HowToUse\\bootloader.bin" "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\HowToUse.ino.bootloader.bin" ) ELSE ( IF EXIST "C:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\hardware\\esp32\\3.0.2\\variants\\esp32\\bootloader.bin" ( COPY "C:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\hardware\\esp32\\3.0.2\\variants\\esp32\\bootloader.bin" "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\HowToUse.ino.bootloader.bin" ) ELSE ( "C:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\tools\\esptool_py\\4.6\\esptool.exe" --chip esp32 elf2image --flash_mode dio --flash_freq 80m --flash_size 4MB -o "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\HowToUse.ino.bootloader.bin" "C:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\tools\\esp32-arduino-libs\\idf-release_v5.1-bd2b9390ef\\esp32\\bin\\bootloader_qio_80m.elf" ) )
esptool.py v4.6
Creating esp32 image...
Merged 1 ELF section
Successfully created esp32 image.
cmd /c if exist "C:\\Users\\cflem\\AppData\\Local\\Temp\\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\\HowToUse\\build_opt.h" COPY /y "C:\\Users\\cflem\\AppData\\Local\\Temp\\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\\HowToUse\\build_opt.h" "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\build_opt.h"
cmd /c if not exist "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\build_opt.h" type nul > "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\build_opt.h"
cmd /c type nul > "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4/file_opts"
Detecting libraries used...
C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp-x32\2302/bin/xtensa-esp32-elf-g++ -c @C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/flags/cpp_flags -w -Os -w -x c++ -E -CC -DF_CPU=240000000L -DARDUINO=10607 -DARDUINO_ESP32_DEV -DARDUINO_ARCH_ESP32 -DARDUINO_BOARD="ESP32_DEV" -DARDUINO_VARIANT="esp32" -DARDUINO_PARTITION_default -DARDUINO_HOST_OS="windows" -DARDUINO_FQBN="esp32:esp32:esp32:UploadSpeed=921600,CPUFreq=240,FlashFreq=80,FlashMode=qio,FlashSize=4M,PartitionScheme=default,DebugLevel=none,PSRAM=disabled,LoopCore=1,EventsCore=1,EraseFlash=none,JTAGAdapter=default,ZigbeeMode=default" -DESP32 -DCORE_DEBUG_LEVEL=0 -DARDUINO_RUNNING_CORE=1 -DARDUINO_EVENT_RUNNING_CORE=1 -DARDUINO_USB_CDC_ON_BOOT=0 @C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/flags/defines -IC:\Users\cflem\AppData\Local\Temp\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\HowToUse -iprefix C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/include/ @C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/flags/includes -IC:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/qio_qspi/include -IC:\Users\cflem\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.2\cores\esp32 -IC:\Users\cflem\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.2\variants\esp32 @C:\Users\cflem\AppData\Local\Temp\arduino\sketches\539CE27B4EEFB81498C741CB6FA3F5E4/build_opt.h @C:\Users\cflem\AppData\Local\Temp\arduino\sketches\539CE27B4EEFB81498C741CB6FA3F5E4/file_opts C:\Users\cflem\AppData\Local\Temp\arduino\sketches\539CE27B4EEFB81498C741CB6FA3F5E4\sketch\HowToUse.ino.cpp -o nul
Alternatives for BasicLinearAlgebra.h: ***@***.***
ResolveLibrary(BasicLinearAlgebra.h)
-> candidates: ***@***.***
C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp-x32\2302/bin/xtensa-esp32-elf-g++ -c @C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/flags/cpp_flags -w -Os -w -x c++ -E -CC -DF_CPU=240000000L -DARDUINO=10607 -DARDUINO_ESP32_DEV -DARDUINO_ARCH_ESP32 -DARDUINO_BOARD="ESP32_DEV" -DARDUINO_VARIANT="esp32" -DARDUINO_PARTITION_default -DARDUINO_HOST_OS="windows" -DARDUINO_FQBN="esp32:esp32:esp32:UploadSpeed=921600,CPUFreq=240,FlashFreq=80,FlashMode=qio,FlashSize=4M,PartitionScheme=default,DebugLevel=none,PSRAM=disabled,LoopCore=1,EventsCore=1,EraseFlash=none,JTAGAdapter=default,ZigbeeMode=default" -DESP32 -DCORE_DEBUG_LEVEL=0 -DARDUINO_RUNNING_CORE=1 -DARDUINO_EVENT_RUNNING_CORE=1 -DARDUINO_USB_CDC_ON_BOOT=0 @C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/flags/defines -IC:\Users\cflem\AppData\Local\Temp\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\HowToUse -iprefix C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/include/ @C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/flags/includes -IC:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/qio_qspi/include -IC:\Users\cflem\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.2\cores\esp32 -IC:\Users\cflem\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.2\variants\esp32 -Id:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra @C:\Users\cflem\AppData\Local\Temp\arduino\sketches\539CE27B4EEFB81498C741CB6FA3F5E4/build_opt.h @C:\Users\cflem\AppData\Local\Temp\arduino\sketches\539CE27B4EEFB81498C741CB6FA3F5E4/file_opts C:\Users\cflem\AppData\Local\Temp\arduino\sketches\539CE27B4EEFB81498C741CB6FA3F5E4\sketch\HowToUse.ino.cpp -o nul
Generating function prototypes...
C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp-x32\2302/bin/xtensa-esp32-elf-g++ -c @C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/flags/cpp_flags -w -Os -w -x c++ -E -CC -DF_CPU=240000000L -DARDUINO=10607 -DARDUINO_ESP32_DEV -DARDUINO_ARCH_ESP32 -DARDUINO_BOARD="ESP32_DEV" -DARDUINO_VARIANT="esp32" -DARDUINO_PARTITION_default -DARDUINO_HOST_OS="windows" -DARDUINO_FQBN="esp32:esp32:esp32:UploadSpeed=921600,CPUFreq=240,FlashFreq=80,FlashMode=qio,FlashSize=4M,PartitionScheme=default,DebugLevel=none,PSRAM=disabled,LoopCore=1,EventsCore=1,EraseFlash=none,JTAGAdapter=default,ZigbeeMode=default" -DESP32 -DCORE_DEBUG_LEVEL=0 -DARDUINO_RUNNING_CORE=1 -DARDUINO_EVENT_RUNNING_CORE=1 -DARDUINO_USB_CDC_ON_BOOT=0 @C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/flags/defines -IC:\Users\cflem\AppData\Local\Temp\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\HowToUse -iprefix C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/include/ @C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/flags/includes -IC:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/qio_qspi/include -IC:\Users\cflem\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.2\cores\esp32 -IC:\Users\cflem\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.2\variants\esp32 -Id:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra @C:\Users\cflem\AppData\Local\Temp\arduino\sketches\539CE27B4EEFB81498C741CB6FA3F5E4/build_opt.h @C:\Users\cflem\AppData\Local\Temp\arduino\sketches\539CE27B4EEFB81498C741CB6FA3F5E4/file_opts C:\Users\cflem\AppData\Local\Temp\arduino\sketches\539CE27B4EEFB81498C741CB6FA3F5E4\sketch\HowToUse.ino.cpp -o C:\Users\cflem\AppData\Local\Temp\1603568161\sketch_merged.cpp
C:\Users\cflem\AppData\Local\Arduino15\packages\builtin\tools\ctags\5.8-arduino11/ctags -u --language-force=c++ -f - --c++-kinds=svpf --fields=KSTtzns --line-directives C:\Users\cflem\AppData\Local\Temp\1603568161\sketch_merged.cpp
Compiling sketch...
"C:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\tools\\esp-x32\\2302/bin/xtensa-esp32-elf-g++" -MMD -c ***@***.***:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\tools\\esp32-arduino-libs\\idf-release_v5.1-bd2b9390ef\\esp32/flags/cpp_flags" -Os -DF_CPU=240000000L -DARDUINO=10607 -DARDUINO_ESP32_DEV -DARDUINO_ARCH_ESP32 "-DARDUINO_BOARD=\"ESP32_DEV\"" "-DARDUINO_VARIANT=\"esp32\"" -DARDUINO_PARTITION_default "-DARDUINO_HOST_OS=\"windows\"" "-DARDUINO_FQBN=\"esp32:esp32:esp32:UploadSpeed=921600,CPUFreq=240,FlashFreq=80,FlashMode=qio,FlashSize=4M,PartitionScheme=default,DebugLevel=none,PSRAM=disabled,LoopCore=1,EventsCore=1,EraseFlash=none,JTAGAdapter=default,ZigbeeMode=default\"" -DESP32 -DCORE_DEBUG_LEVEL=0 -DARDUINO_RUNNING_CORE=1 -DARDUINO_EVENT_RUNNING_CORE=1 -DARDUINO_USB_CDC_ON_BOOT=0 ***@***.***:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\tools\\esp32-arduino-libs\\idf-release_v5.1-bd2b9390ef\\esp32/flags/defines" "-IC:\\Users\\cflem\\AppData\\Local\\Temp\\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\\HowToUse" -iprefix "C:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\tools\\esp32-arduino-libs\\idf-release_v5.1-bd2b9390ef\\esp32/include/" ***@***.***:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\tools\\esp32-arduino-libs\\idf-release_v5.1-bd2b9390ef\\esp32/flags/includes" "-IC:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\tools\\esp32-arduino-libs\\idf-release_v5.1-bd2b9390ef\\esp32/qio_qspi/include" "-IC:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\hardware\\esp32\\3.0.2\\cores\\esp32" "-IC:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\hardware\\esp32\\3.0.2\\variants\\esp32" "-Id:\\Users\\cflem\\Documents\\Arduino\\libraries\\BasicLinearAlgebra" ***@***.***:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4/build_opt.h" ***@***.***:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4/file_opts" "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\sketch\\HowToUse.ino.cpp" -o "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\sketch\\HowToUse.ino.cpp.o"
In file included from d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/BasicLinearAlgebra.h:10,
from C:\Users\cflem\AppData\Local\Temp\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\HowToUse\HowToUse.ino:1:
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:136:43: error: expected unqualified-id before 'const'
136 | HorizontalConcat<LeftType, RightType>(const LeftType &l, const RightType &r) : left(l), right(r) {}
| ^~~~~
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:136:43: error: expected ')' before 'const'
136 | HorizontalConcat<LeftType, RightType>(const LeftType &l, const RightType &r) : left(l), right(r) {}
| ~^~~~~
| )
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:151:41: error: expected unqualified-id before 'const'
151 | VerticalConcat<TopType, BottomType>(const TopType &t, const BottomType &b) : top(t), bottom(b) {}
| ^~~~~
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:151:41: error: expected ')' before 'const'
151 | VerticalConcat<TopType, BottomType>(const TopType &t, const BottomType &b) : top(t), bottom(b) {}
| ~^~~~~
| )
In file included from d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/BasicLinearAlgebra.h:181:
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/impl/BasicLinearAlgebra.h: In instantiation of 'BLA::HorizontalConcat<DerivedType, OperandType> BLA::operator||(const MatrixBase<DerivedType, Rows, DerivedType::Cols, DType>&, const MatrixBase<OperandType, Rows, OperandType::Cols, DType>&) [with DerivedType = Matrix<3, 3>; OperandType = Matrix<3, 3>; int Rows = 3; DType = float]':
C:\Users\cflem\AppData\Local\Temp\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\HowToUse\HowToUse.ino:70:39: required from here
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/impl/BasicLinearAlgebra.h:208:12: error: no matching function for call to 'BLA::HorizontalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >::HorizontalConcat(const BLA::Matrix<3, 3>&, const BLA::Matrix<3, 3>&)'
208 | return HorizontalConcat<DerivedType, OperandType>(static_cast<const DerivedType &>(left),
| ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
209 | static_cast<const OperandType &>(right));
| ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:130:8: note: candidate: 'constexpr BLA::HorizontalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >::HorizontalConcat(const BLA::HorizontalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >&)'
130 | struct HorizontalConcat : public MatrixBase<HorizontalConcat<LeftType, RightType>, LeftType::Rows,
| ^~~~~~~~~~~~~~~~
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:130:8: note: candidate expects 1 argument, 2 provided
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:130:8: note: candidate: 'constexpr BLA::HorizontalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >::HorizontalConcat(BLA::HorizontalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >&&)'
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:130:8: note: candidate expects 1 argument, 2 provided
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/impl/BasicLinearAlgebra.h: In instantiation of 'BLA::VerticalConcat<DerivedType, OperandType> BLA::operator&&(const MatrixBase<DerivedType, DerivedType::Rows, Cols, DType>&, const MatrixBase<OperandType, OperandType::Rows, Cols, DType>&) [with DerivedType = Matrix<3, 3>; OperandType = Matrix<3, 3>; int Cols = 3; DType = float]':
C:\Users\cflem\AppData\Local\Temp\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\HowToUse\HowToUse.ino:73:40: required from here
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/impl/BasicLinearAlgebra.h:218:12: error: no matching function for call to 'BLA::VerticalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >::VerticalConcat(const BLA::Matrix<3, 3>&, const BLA::Matrix<3, 3>&)'
218 | return VerticalConcat<DerivedType, OperandType>(static_cast<const DerivedType &>(top),
| ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
219 | static_cast<const OperandType &>(bottom));
| ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:145:8: note: candidate: 'constexpr BLA::VerticalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >::VerticalConcat(const BLA::VerticalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >&)'
145 | struct VerticalConcat : public MatrixBase<VerticalConcat<TopType, BottomType>, TopType::Rows + BottomType::Rows,
| ^~~~~~~~~~~~~~
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:145:8: note: candidate expects 1 argument, 2 provided
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:145:8: note: candidate: 'constexpr BLA::VerticalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >::VerticalConcat(BLA::VerticalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >&&)'
d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:145:8: note: candidate expects 1 argument, 2 provided

Using library BasicLinearAlgebra at version 5.0 in folder: D:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra
exit status 1
Compilation error: exit status 1

Best regards,
Chip Fleming

Please use ***@***.*** when addressing email messages to me, as it's my intended "permanent" address. But a direct reply will also work, for now.

On Sunday, July 7th, 2024 at 9:35 PM, tomstewart89 ***@***.***> wrote:

> Hi ***@***.***(https://github.com/CF20852) , thanks for opening this issue. It's a bit tricky to understand what the source of the problem is from those compiler errors. Can you tell me some more about how you're compiling the library? Are you using the arduino IDE? If so, what version and also what version?
>
> Also, I see you're compiling the HowToUse sketch, did you modify it in any way?
>
> Thanks again!
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/tomstewart89/BasicLinearAlgebra/issues/82#issuecomment-2213008687), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/ACBERCLF5IPRVSTHVVLFFULZLIJJRAVCNFSM6AAAAABKPTLKCOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDEMJTGAYDQNRYG4).
> You are receiving this because you were mentioned.Message ID: ***@***.***>

Tom, I lied.

I changed N to 6:
#defineN (6)

It seems to compile successfully with N = 3.

I want to use your library to solve Eq. 7 in this TI appnote, with six input points:
https://www.ti.com/lit/an/slyt277/slyt277.pdf

Sorry, sorry, sorry,...
Chip Fleming

Please use ***@***.*** when addressing email messages to me, as it's my intended "permanent" address. But a direct reply will also work, for now.

On Monday, July 8th, 2024 at 1:38 AM, Chip Fleming ***@***.***> wrote:

> Tom,
>
> Here's the complete output of the Arduino 2.3.2 IDE with verbose compiler messages:
>
> FQBN: esp32:esp32:esp32
> Using board 'esp32' from platform in folder: C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.2
> Using core 'esp32' from platform in folder: C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.2
>
> cmd /c if exist "C:\\Users\\cflem\\AppData\\Local\\Temp\\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\\HowToUse\\partitions.csv" COPY /y "C:\\Users\\cflem\\AppData\\Local\\Temp\\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\\HowToUse\\partitions.csv" "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\partitions.csv"
> cmd /c if not exist "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\partitions.csv" if exist "C:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\hardware\\esp32\\3.0.2\\variants\\esp32\\partitions.csv" COPY "C:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\hardware\\esp32\\3.0.2\\variants\\esp32\\partitions.csv" "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\partitions.csv"
> cmd /c if not exist "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\partitions.csv" COPY "C:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\hardware\\esp32\\3.0.2\\tools\\partitions\\default.csv" "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\partitions.csv"
> 1 file(s) copied.
> cmd /c IF EXIST "C:\\Users\\cflem\\AppData\\Local\\Temp\\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\\HowToUse\\bootloader.bin" ( COPY /y "C:\\Users\\cflem\\AppData\\Local\\Temp\\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\\HowToUse\\bootloader.bin" "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\HowToUse.ino.bootloader.bin" ) ELSE ( IF EXIST "C:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\hardware\\esp32\\3.0.2\\variants\\esp32\\bootloader.bin" ( COPY "C:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\hardware\\esp32\\3.0.2\\variants\\esp32\\bootloader.bin" "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\HowToUse.ino.bootloader.bin" ) ELSE ( "C:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\tools\\esptool_py\\4.6\\esptool.exe" --chip esp32 elf2image --flash_mode dio --flash_freq 80m --flash_size 4MB -o "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\HowToUse.ino.bootloader.bin" "C:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\tools\\esp32-arduino-libs\\idf-release_v5.1-bd2b9390ef\\esp32\\bin\\bootloader_qio_80m.elf" ) )
> esptool.py v4.6
> Creating esp32 image...
> Merged 1 ELF section
> Successfully created esp32 image.
> cmd /c if exist "C:\\Users\\cflem\\AppData\\Local\\Temp\\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\\HowToUse\\build_opt.h" COPY /y "C:\\Users\\cflem\\AppData\\Local\\Temp\\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\\HowToUse\\build_opt.h" "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\build_opt.h"
> cmd /c if not exist "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\build_opt.h" type nul > "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\build_opt.h"
> cmd /c type nul > "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4/file_opts"
> Detecting libraries used...
> C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp-x32\2302/bin/xtensa-esp32-elf-g++ -c @C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/flags/cpp_flags -w -Os -w -x c++ -E -CC -DF_CPU=240000000L -DARDUINO=10607 -DARDUINO_ESP32_DEV -DARDUINO_ARCH_ESP32 -DARDUINO_BOARD="ESP32_DEV" -DARDUINO_VARIANT="esp32" -DARDUINO_PARTITION_default -DARDUINO_HOST_OS="windows" -DARDUINO_FQBN="esp32:esp32:esp32:UploadSpeed=921600,CPUFreq=240,FlashFreq=80,FlashMode=qio,FlashSize=4M,PartitionScheme=default,DebugLevel=none,PSRAM=disabled,LoopCore=1,EventsCore=1,EraseFlash=none,JTAGAdapter=default,ZigbeeMode=default" -DESP32 -DCORE_DEBUG_LEVEL=0 -DARDUINO_RUNNING_CORE=1 -DARDUINO_EVENT_RUNNING_CORE=1 -DARDUINO_USB_CDC_ON_BOOT=0 @C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/flags/defines -IC:\Users\cflem\AppData\Local\Temp\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\HowToUse -iprefix C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/include/ @C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/flags/includes -IC:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/qio_qspi/include -IC:\Users\cflem\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.2\cores\esp32 -IC:\Users\cflem\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.2\variants\esp32 @C:\Users\cflem\AppData\Local\Temp\arduino\sketches\539CE27B4EEFB81498C741CB6FA3F5E4/build_opt.h @C:\Users\cflem\AppData\Local\Temp\arduino\sketches\539CE27B4EEFB81498C741CB6FA3F5E4/file_opts C:\Users\cflem\AppData\Local\Temp\arduino\sketches\539CE27B4EEFB81498C741CB6FA3F5E4\sketch\HowToUse.ino.cpp -o nul
> Alternatives for BasicLinearAlgebra.h: ***@***.***
> ResolveLibrary(BasicLinearAlgebra.h)
> -> candidates: ***@***.***
> C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp-x32\2302/bin/xtensa-esp32-elf-g++ -c @C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/flags/cpp_flags -w -Os -w -x c++ -E -CC -DF_CPU=240000000L -DARDUINO=10607 -DARDUINO_ESP32_DEV -DARDUINO_ARCH_ESP32 -DARDUINO_BOARD="ESP32_DEV" -DARDUINO_VARIANT="esp32" -DARDUINO_PARTITION_default -DARDUINO_HOST_OS="windows" -DARDUINO_FQBN="esp32:esp32:esp32:UploadSpeed=921600,CPUFreq=240,FlashFreq=80,FlashMode=qio,FlashSize=4M,PartitionScheme=default,DebugLevel=none,PSRAM=disabled,LoopCore=1,EventsCore=1,EraseFlash=none,JTAGAdapter=default,ZigbeeMode=default" -DESP32 -DCORE_DEBUG_LEVEL=0 -DARDUINO_RUNNING_CORE=1 -DARDUINO_EVENT_RUNNING_CORE=1 -DARDUINO_USB_CDC_ON_BOOT=0 @C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/flags/defines -IC:\Users\cflem\AppData\Local\Temp\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\HowToUse -iprefix C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/include/ @C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/flags/includes -IC:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/qio_qspi/include -IC:\Users\cflem\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.2\cores\esp32 -IC:\Users\cflem\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.2\variants\esp32 -Id:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra @C:\Users\cflem\AppData\Local\Temp\arduino\sketches\539CE27B4EEFB81498C741CB6FA3F5E4/build_opt.h @C:\Users\cflem\AppData\Local\Temp\arduino\sketches\539CE27B4EEFB81498C741CB6FA3F5E4/file_opts C:\Users\cflem\AppData\Local\Temp\arduino\sketches\539CE27B4EEFB81498C741CB6FA3F5E4\sketch\HowToUse.ino.cpp -o nul
> Generating function prototypes...
> C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp-x32\2302/bin/xtensa-esp32-elf-g++ -c @C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/flags/cpp_flags -w -Os -w -x c++ -E -CC -DF_CPU=240000000L -DARDUINO=10607 -DARDUINO_ESP32_DEV -DARDUINO_ARCH_ESP32 -DARDUINO_BOARD="ESP32_DEV" -DARDUINO_VARIANT="esp32" -DARDUINO_PARTITION_default -DARDUINO_HOST_OS="windows" -DARDUINO_FQBN="esp32:esp32:esp32:UploadSpeed=921600,CPUFreq=240,FlashFreq=80,FlashMode=qio,FlashSize=4M,PartitionScheme=default,DebugLevel=none,PSRAM=disabled,LoopCore=1,EventsCore=1,EraseFlash=none,JTAGAdapter=default,ZigbeeMode=default" -DESP32 -DCORE_DEBUG_LEVEL=0 -DARDUINO_RUNNING_CORE=1 -DARDUINO_EVENT_RUNNING_CORE=1 -DARDUINO_USB_CDC_ON_BOOT=0 @C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/flags/defines -IC:\Users\cflem\AppData\Local\Temp\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\HowToUse -iprefix C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/include/ @C:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/flags/includes -IC:\Users\cflem\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-bd2b9390ef\esp32/qio_qspi/include -IC:\Users\cflem\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.2\cores\esp32 -IC:\Users\cflem\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.2\variants\esp32 -Id:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra @C:\Users\cflem\AppData\Local\Temp\arduino\sketches\539CE27B4EEFB81498C741CB6FA3F5E4/build_opt.h @C:\Users\cflem\AppData\Local\Temp\arduino\sketches\539CE27B4EEFB81498C741CB6FA3F5E4/file_opts C:\Users\cflem\AppData\Local\Temp\arduino\sketches\539CE27B4EEFB81498C741CB6FA3F5E4\sketch\HowToUse.ino.cpp -o C:\Users\cflem\AppData\Local\Temp\1603568161\sketch_merged.cpp
> C:\Users\cflem\AppData\Local\Arduino15\packages\builtin\tools\ctags\5.8-arduino11/ctags -u --language-force=c++ -f - --c++-kinds=svpf --fields=KSTtzns --line-directives C:\Users\cflem\AppData\Local\Temp\1603568161\sketch_merged.cpp
> Compiling sketch...
> "C:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\tools\\esp-x32\\2302/bin/xtensa-esp32-elf-g++" -MMD -c ***@***.***:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\tools\\esp32-arduino-libs\\idf-release_v5.1-bd2b9390ef\\esp32/flags/cpp_flags" -Os -DF_CPU=240000000L -DARDUINO=10607 -DARDUINO_ESP32_DEV -DARDUINO_ARCH_ESP32 "-DARDUINO_BOARD=\"ESP32_DEV\"" "-DARDUINO_VARIANT=\"esp32\"" -DARDUINO_PARTITION_default "-DARDUINO_HOST_OS=\"windows\"" "-DARDUINO_FQBN=\"esp32:esp32:esp32:UploadSpeed=921600,CPUFreq=240,FlashFreq=80,FlashMode=qio,FlashSize=4M,PartitionScheme=default,DebugLevel=none,PSRAM=disabled,LoopCore=1,EventsCore=1,EraseFlash=none,JTAGAdapter=default,ZigbeeMode=default\"" -DESP32 -DCORE_DEBUG_LEVEL=0 -DARDUINO_RUNNING_CORE=1 -DARDUINO_EVENT_RUNNING_CORE=1 -DARDUINO_USB_CDC_ON_BOOT=0 ***@***.***:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\tools\\esp32-arduino-libs\\idf-release_v5.1-bd2b9390ef\\esp32/flags/defines" "-IC:\\Users\\cflem\\AppData\\Local\\Temp\\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\\HowToUse" -iprefix "C:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\tools\\esp32-arduino-libs\\idf-release_v5.1-bd2b9390ef\\esp32/include/" ***@***.***:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\tools\\esp32-arduino-libs\\idf-release_v5.1-bd2b9390ef\\esp32/flags/includes" "-IC:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\tools\\esp32-arduino-libs\\idf-release_v5.1-bd2b9390ef\\esp32/qio_qspi/include" "-IC:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\hardware\\esp32\\3.0.2\\cores\\esp32" "-IC:\\Users\\cflem\\AppData\\Local\\Arduino15\\packages\\esp32\\hardware\\esp32\\3.0.2\\variants\\esp32" "-Id:\\Users\\cflem\\Documents\\Arduino\\libraries\\BasicLinearAlgebra" ***@***.***:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4/build_opt.h" ***@***.***:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4/file_opts" "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\sketch\\HowToUse.ino.cpp" -o "C:\\Users\\cflem\\AppData\\Local\\Temp\\arduino\\sketches\\539CE27B4EEFB81498C741CB6FA3F5E4\\sketch\\HowToUse.ino.cpp.o"
> In file included from d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/BasicLinearAlgebra.h:10,
> from C:\Users\cflem\AppData\Local\Temp\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\HowToUse\HowToUse.ino:1:
> d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:136:43: error: expected unqualified-id before 'const'
> 136 | HorizontalConcat<LeftType, RightType>(const LeftType &l, const RightType &r) : left(l), right(r) {}
> | ^~~~~
> d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:136:43: error: expected ')' before 'const'
> 136 | HorizontalConcat<LeftType, RightType>(const LeftType &l, const RightType &r) : left(l), right(r) {}
> | ~^~~~~
> | )
> d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:151:41: error: expected unqualified-id before 'const'
> 151 | VerticalConcat<TopType, BottomType>(const TopType &t, const BottomType &b) : top(t), bottom(b) {}
> | ^~~~~
> d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:151:41: error: expected ')' before 'const'
> 151 | VerticalConcat<TopType, BottomType>(const TopType &t, const BottomType &b) : top(t), bottom(b) {}
> | ~^~~~~
> | )
> In file included from d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/BasicLinearAlgebra.h:181:
> d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/impl/BasicLinearAlgebra.h: In instantiation of 'BLA::HorizontalConcat<DerivedType, OperandType> BLA::operator||(const MatrixBase<DerivedType, Rows, DerivedType::Cols, DType>&, const MatrixBase<OperandType, Rows, OperandType::Cols, DType>&) [with DerivedType = Matrix<3, 3>; OperandType = Matrix<3, 3>; int Rows = 3; DType = float]':
> C:\Users\cflem\AppData\Local\Temp\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\HowToUse\HowToUse.ino:70:39: required from here
> d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/impl/BasicLinearAlgebra.h:208:12: error: no matching function for call to 'BLA::HorizontalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >::HorizontalConcat(const BLA::Matrix<3, 3>&, const BLA::Matrix<3, 3>&)'
> 208 | return HorizontalConcat<DerivedType, OperandType>(static_cast<const DerivedType &>(left),
> | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 209 | static_cast<const OperandType &>(right));
> | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:130:8: note: candidate: 'constexpr BLA::HorizontalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >::HorizontalConcat(const BLA::HorizontalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >&)'
> 130 | struct HorizontalConcat : public MatrixBase<HorizontalConcat<LeftType, RightType>, LeftType::Rows,
> | ^~~~~~~~~~~~~~~~
> d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:130:8: note: candidate expects 1 argument, 2 provided
> d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:130:8: note: candidate: 'constexpr BLA::HorizontalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >::HorizontalConcat(BLA::HorizontalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >&&)'
> d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:130:8: note: candidate expects 1 argument, 2 provided
> d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/impl/BasicLinearAlgebra.h: In instantiation of 'BLA::VerticalConcat<DerivedType, OperandType> BLA::operator&&(const MatrixBase<DerivedType, DerivedType::Rows, Cols, DType>&, const MatrixBase<OperandType, OperandType::Rows, Cols, DType>&) [with DerivedType = Matrix<3, 3>; OperandType = Matrix<3, 3>; int Cols = 3; DType = float]':
> C:\Users\cflem\AppData\Local\Temp\.arduinoIDE-unsaved202467-3320-15y8msi.qryx\HowToUse\HowToUse.ino:73:40: required from here
> d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/impl/BasicLinearAlgebra.h:218:12: error: no matching function for call to 'BLA::VerticalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >::VerticalConcat(const BLA::Matrix<3, 3>&, const BLA::Matrix<3, 3>&)'
> 218 | return VerticalConcat<DerivedType, OperandType>(static_cast<const DerivedType &>(top),
> | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 219 | static_cast<const OperandType &>(bottom));
> | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:145:8: note: candidate: 'constexpr BLA::VerticalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >::VerticalConcat(const BLA::VerticalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >&)'
> 145 | struct VerticalConcat : public MatrixBase<VerticalConcat<TopType, BottomType>, TopType::Rows + BottomType::Rows,
> | ^~~~~~~~~~~~~~
> d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:145:8: note: candidate expects 1 argument, 2 provided
> d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:145:8: note: candidate: 'constexpr BLA::VerticalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >::VerticalConcat(BLA::VerticalConcat<BLA::Matrix<3, 3>, BLA::Matrix<3, 3> >&&)'
> d:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra/ElementStorage.h:145:8: note: candidate expects 1 argument, 2 provided
>
> Using library BasicLinearAlgebra at version 5.0 in folder: D:\Users\cflem\Documents\Arduino\libraries\BasicLinearAlgebra
> exit status 1
> Compilation error: exit status 1
>
> Best regards,
> Chip Fleming
>
> Please use ***@***.*** when addressing email messages to me, as it's my intended "permanent" address. But a direct reply will also work, for now.
>
> On Sunday, July 7th, 2024 at 9:35 PM, tomstewart89 ***@***.***> wrote:
>
>> Hi ***@***.***(https://github.com/CF20852) , thanks for opening this issue. It's a bit tricky to understand what the source of the problem is from those compiler errors. Can you tell me some more about how you're compiling the library? Are you using the arduino IDE? If so, what version and also what version?
>>
>> Also, I see you're compiling the HowToUse sketch, did you modify it in any way?
>>
>> Thanks again!
>>
>> —
>> Reply to this email directly, [view it on GitHub](https://github.com/tomstewart89/BasicLinearAlgebra/issues/82#issuecomment-2213008687), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/ACBERCLF5IPRVSTHVVLFFULZLIJJRAVCNFSM6AAAAABKPTLKCOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDEMJTGAYDQNRYG4).
>> You are receiving this because you were mentioned.Message ID: ***@***.***>

--- a/ElementStorage.h
+++ b/ElementStorage.h
@@ -133,7 +133,7 @@ struct HorizontalConcat : public MatrixBase<HorizontalConcat<LeftType, RightType
     const LeftType &left;
     const RightType &right;
 
-    HorizontalConcat<LeftType, RightType>(const LeftType &l, const RightType &r) : left(l), right(r) {}
+    HorizontalConcat(const LeftType &l, const RightType &r) : left(l), right(r) {}
 
     typename LeftType::DType operator()(int row, int col) const
     {
@@ -148,7 +148,7 @@ struct VerticalConcat : public MatrixBase<VerticalConcat<TopType, BottomType>, T
     const TopType &top;
     const BottomType &bottom;
 
-    VerticalConcat<TopType, BottomType>(const TopType &t, const BottomType &b) : top(t), bottom(b) {}
+    VerticalConcat(const TopType &t, const BottomType &b) : top(t), bottom(b) {}
 
     typename TopType::DType operator()(int row, int col) const
     {

--- a/library.properties
+++ b/library.properties
@@ -1,5 +1,5 @@
 name=BasicLinearAlgebra
-version=5.0
+version=5.1
 author=Tom Stewart <tomstewart89@hotmail.com>
 maintainer=Tom Stewart <tomstewart89@hotmail.com>
 sentence=A library for representing matrices and doing matrix math on arduino 

